overlord/devicestate/firstboot.go

```diff
@@ -122,20 +116,25 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 	timings.Run(tm, "import-assertions", "import assertions from seed", func(nested timings.Measurer) {
 		_, err = importAssertionsFromSeed(st, deviceSeed)
 	})
-	if err == errNothingToDo {
-		return trivialSeeding(st, markSeeded), nil
-	}
-	if err != nil {
+	if err != nil && err != errNothingToDo {
 		return nil, err
 	}
 
+	createMarkSeeded := func() *state.Task {
```

Comment:
if we have to do this, maybe just have a global markSeededTask(st) function, and stop passing it in trivialSeeding?

Reply:
Sounds good, thanks.